xylib/bruker_raw.cpp

```diff
@@ -305,6 +312,178 @@ void BrukerRawDataSet::load_version1_01(std::istream &f)
+    while (1 == 1) {
+        segment_type = read_uint32_le(f);       // offset +0
+        if (segment_type == 0 || segment_type == 160)
+            break; // start of ranges
+        segment_len = read_uint32_le(f);        // offset +4
+        assert(segment_len >= 8);
+        if (segment_type == 10) { // VarInfo
+            assert(segment_len >= 36);
+            f.ignore(4);                        // offset +8
+            string tag_name = read_string(f, 24);      // offset +12
+            meta[tag_name] = read_string(f, segment_len-36); // offset +36
+        }
+        else if (segment_type == 30) { // HardwareConfiguration
+            assert(segment_len >= 120);
+            f.ignore(64);                               // offset +8
+            meta["ALPHA_AVERAGE"] = S(read_dbl_le(f));  // offset +72
+            meta["ALPHA1"] = S(read_dbl_le(f));         // offset +80
+            meta["ALPHA2"] = S(read_dbl_le(f));         // offset +88
+            meta["BETA"] = S(read_dbl_le(f));           // offset +96
+            meta["ALPHA_RATIO"] = S(read_dbl_le(f));    // offset +104
+            f.ignore(4);                                // offset +112
+            meta["ANODE_MATERIAL"] = read_string(f,4);  // offset +116
+            f.ignore(segment_len-120);                  // offset +120
+        }
+        else if (segment_type == 60) { // DriveAlignment
+            assert(segment_len >= 76);
+            int align_flag = read_uint32_le(f);         // offset +8
+            string drive_name = read_string(f, 24);     // offset +12
+            f.ignore(32);                               // offset +36
+            double delta = read_dbl_le(f);              // offset +68
+            f.ignore(segment_len-76);
+            meta["DRIVE" + S(drive_num) + "_ALIGN_FLAG"] = S(align_flag);
+            meta["DRIVE" + S(drive_num) + "_DELTA"] = S(delta);
+            drive_num++;
+        }
+        else { // skip others
+            f.ignore(segment_len-8);
+        }
+    }
+
+    // now process ranges
+    while (segment_type == 0 || segment_type == 160) {
+        Block* blk = new Block;
+        // primary range header
+        f.ignore(28);                               // offset +4
+        blk->meta["SCAN_TYPE"] = read_string(f,24); // offset +32
+        f.ignore(16);                               // offset +56
+        double start_angle = read_dbl_le(f);        // offset +72
+        blk->meta["START_ANGLE"] = S(start_angle);
+        double step_size = read_dbl_le(f);          // offset +80
+        blk->meta["STEP_SIZE"] = S(step_size);
+        int steps = read_uint32_le(f);              // offset +88
+        blk->meta["STEPS"] = S(steps);
+        blk->meta["TIME_PER_STEP"] = S(read_flt_le(f)); // offset +92
+        f.ignore(4);                                // offset +96
+        blk->meta["GENERATOR_VOLTAGE"] = Su(read_flt_le(f)); // +100
+        blk->meta["GENERATOR_CURRENT"] = Su(read_flt_le(f)); // +104
```

Comment:
Was it intentional that `GENERATOR_VOLTAGE` and `GENERATOR_CURRENT` are read as float numbers and then formatted as integers?

Reply:
It was intentional. The voltage and current are stored as single precision floating point numbers now but, to my knowledge, the generators require integral values for tube kV and mA. But really if you think it should be S instead of Su, that would seem to be fine.